ui/src/app/app.service.ts

```diff
@@ -103,19 +103,30 @@ export class AppService {
                 return;
             }
 
-
             let pipKey = lastUpdate.key + '-' + lastUpdate.name;
             if (!pips.get(pipKey)) {
                 return;
             }
 
-            if (pips.get(pipKey).last_modified < lastUpdate.last_modified) {
+            if ((new Date(pips.get(pipKey).last_modified)).getTime() < lastUpdate.last_modified * 1000) {
                 let params = this._routerService.getRouteParams({}, this._routeActivated);
 
-                if (params['key'] && params['key'] === lastUpdate.key && params['pipName'] === lastUpdate.name) {
-                    this._pipStore.externalModification(pipKey);
+                if (!params['key'] || params['key'] !== lastUpdate.key) {
+                    return;
+                }
+
+                // mark to update applications linked to this pipeline
+                this._pipStore.getPipelineResolver(params['key'], lastUpdate.name)
+                    .subscribe((pip) => {
+                        if (pip && Array.isArray(pip.attached_application)) {
+                            pip.attached_application.forEach((app) => this._appStore.markUpdate(params['key'], app.name));
```

Comment:
I think:
1. Remove toUpdate + markToUpdate
2. Remove from cache LinkedIn app if appName IS not on the current route else resync it

That all